DESIGN.md

```diff
@@ -0,0 +1,27 @@
+## Overview
+
+### Target Audience
+
+- Developer A: high-level framework writer
+  - wants to provide barebones initial configuration
+  - wants to consume a simple event stream
+  - wants to inject events into a managed event loop (easy state machine semantics)
+
+```go
+c := client.New(connectionOptions..) // package scheduler/client
+eventCh, errCh := c.Register(registrationOptions..)
```

Comment:
+1

The original idea was to have schedulers just decode events in a for loop. I realised I had some lingering code in my computer which I just pushed to https://github.com/mesos/mesos-go/pull/211